MetaMask/metamask-extension
# Feat/19274/ds popover update account list menu

## Explanation

* Fixes #19274 

Goal: Replace a prod component to use the DS Popover as a stress test use case. 

Results: In the end was successful on replacing AccountListMenu's Menu component to use Popover. Upon replacing I noticed an accessibility issue that occurs from using portals where the tab index of the popover item is at the bottom rather than being in line. To fix this we utilize the ModalFocus component to insert the popover into the indexing and then we track as the user tabs through the MenuItems that when they get to the last one it will close the popover and move to the next focus item of the DOM. 

Clicking outside of the component is currently not built into Popover, so that functionality is added for these use cases. In the event we see more reason to add this functionality to Popover then we will make the appropriate updates. 


## Screenshots/Screencaps



### Before

https://www.loom.com/share/05eb9cd959144cb6911a96fd5a347296



### After

https://www.loom.com/share/40e8c829b75d40e19adf2c4a4eb779d1


## Manual Testing Steps

Run `MULTICHAIN=1 yarn start` and use the tab key to navigate to Account List and select the options. (see above videos)



## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
